src/se/terhol/pisemka31/IceCream.java

```diff
@@ -6,13 +6,13 @@
  * @author Radek Oslejsek
  * @version 2010-12-18
  */
-public class IceCream {
+public class IceCream implements Comparable {
```

Comment:
You miss a generic type here, i.e. `Comparable<E>`

Reply:
Done